main.tf

```diff
@@ -1,29 +1,50 @@
-locals {
-  provision_namespace = var.provision_namespace
-  create_package      = var.create_package
+#####################################################
+# IBM Cloud Function Module
+# Copyright 2021 IBM
+#####################################################
+
+data "ibm_resource_group" "resource_group" {
+  name = var.resource_group_name
+}
+
+###################################################################
+# Create Namespace
+###################################################################
+data "ibm_function_namespace" "function_namespace" {
+  count = var.is_new_namespace ? 0 : 1
+  name  = var.namespace_name
 }
 
 resource "ibm_function_namespace" "function_namespace" {
-  count             = var.provision_namespace ? 1 : 0
+  count             = var.is_new_namespace ? 1 : 0
   name              = var.namespace_name
   description       = (var.namespace_description != null ? var.namespace_description : null)
-  resource_group_id = var.resource_group_id
+  resource_group_id = data.ibm_resource_group.resource_group.id
 }
 
+###################################################################
+# Create Package
+###################################################################
+
 resource "ibm_function_package" "function_package" {
-  count                    = var.create_package ? 1 : 0
+  count                    = var.is_new_package ? 1 : 0
   name                     = var.package_name
-  namespace                = var.namespace_name
+  namespace                = var.is_new_namespace ? ibm_function_namespace.function_namespace.0.name : data.ibm_function_namespace.function_namespace.0.name
```

Comment:
we can assign even existing package also an action

Reply:
Fixed